### DISCUSSION TITLE
Load Balancer - How

### DISCUSSION THREAD

**Discussion by jfmatth** [siderolabs/talos#9959] [Q&A]
Just got my Talos cluster running, and LOVING how easy it is to setup - Thank you for this project!

I am a bit lost on how to setup an ingress / Load Balancer for the cluster.  I think it's using the new API Gateway interface, but I am just a little lost.

Any good articles or documentation on setting up your basic inbound Ingress like the 'old days' before API Gateway?

Thanks again for this wonderful project.

**Comment by smira** [comment]
I think we have a list of options in the [documentation](https://www.talos.dev/v1.8/introduction/prodnotes/#decide-the-kubernetes-endpoint). Please let me know if it doesn't answer your question.

**Reply by jfmatth** [reply]
@smira Thank you for the quick response this AM (Cali time).

I have the API working with LB, but wondering if that's all I need to be able to use the Gateway API for inbound HTTP?

I think I tried using the gateway API and was told the CRD's aren't installed, but like I mentioned, new to this project w/o having an LB included.  I'm used to running K3s.

Thanks again, wonderful project.

**Reply by smira** [reply]
Talos ships vanilla Kubernetes, [Gateway API](https://kubernetes.io/docs/concepts/services-networking/gateway/) is available as well. 

You might need to install an implementation of it for your environment, whatever fits your goals. Talos doesn't ship any by default, so it's your choice to use one or another.

**Reply by jfmatth** [reply]
Thanks again.

Are there any CRD's that need to be installed?  Do you know of any good guides helping me get there?

Appreciate the quick responses too.

**Reply by smira** [reply]
If you look into the docs I mentioned, [last section](https://kubernetes.io/docs/concepts/services-networking/gateway/#what-s-next) has instructions.

**Reply by jfmatth** [reply]
Thanks again @smira for the quick attention to this.  Have a wonderful holiday season and day.

**Reply by jfmatth** [reply]
The missing piece was MetalLB, once I installed that it all worked.
